core/src/main/scala/io/finch/request/package.scala

```diff
@@ -49,14 +129,22 @@ package object request {
     def apply[Req](req: Req)(implicit ev: Req => HttpRequest): Future[A]
 
     def flatMap[B](fn: A => RequestReader[B]) = new RequestReader[B] {
+      val item = MultipleItems
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) = self(req) flatMap { fn(_)(req) }
     }
 
     def map[B](fn: A => B) = new RequestReader[B] {
+      val item = self.item
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) = self(req) map fn
     }
     
+    def mapFuture[B](fn: A => Future[B]) = new RequestReader[B] {
```

Comment:
I had to add a similar method into `Router`. I called it `mabeMap`. It operates a different type: `Option` vs. `Future`, which is still `F[_]`. I think we should synchronise our names. I made some research on how this method should be named. A request reader, as well as router are particular cases of Kleisli arrow, with output higher kinded type fixed. It usually called just as `flatMap` or `andThen` (see [cat's example](https://github.com/non/cats/blob/master/core/src/main/scala/cats/arrow/Kleisli.scala#L33)). Do you think we both can go with just `flatMap` then? I can rename it in `Router`.